fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryManager.java

```diff
@@ -123,7 +123,7 @@ public CompletableFuture<Void> mergeIndex(LuceneAnalyzerWrapper analyzerWrapper)
 
     private void mergeIndex(LuceneAnalyzerWrapper analyzerWrapper, Tuple groupingKey) {
         try {
-            getDirectoryWrapper(groupingKey).mergeIndex(analyzerWrapper);
+            getDirectoryWrapper(groupingKey, true).mergeIndex(analyzerWrapper);
```

Comment:
Does this make it so that if you have multiple groupingKeys it will always create a new transaction for each groupingKey, right?

Reply:
Yes. 
And probably more than one, as the fileLock's flush will cause each AgilityContext to commit. 
However, since this merge is always a background operation, it seems reasonable. Doesn't it?

Reply:
Yeah, I think that should be fine for now. There might be an argument in the future for pulling the agility context up, so that it is provided the `FDBDirectory`, and then the same one could be used across groups, only committing as needed.